Sources/NostrSDK/Events/Calendars/TimeBasedCalendarEventNostrEvent.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  TimeBasedCalendarEventEvent.swift
+//
+//
+//  Created by Terry Yiu on 11/16/23.
+//
+
+import Foundation
+
+/// Time-based calendar event spans between a start time and end time.
+public final class TimeBasedCalendarEventNostrEvent: NostrEvent, CalendarEventParticipantInterpreting, HashtagInterpreting, ReferenceTagInterpreting {
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    public init(content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: .timeBasedCalendarEvent, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    /// Universally unique identifier (UUID).
+    public var uuid: String? {
```

Comment:
For the functions and computed vars that are the same in both types...

Maybe could have a parent class called `CalendarEvent` that has the shared code? Just a suggestion, because sometimes this introduces more trouble. Or maybe it could be implemented as a protocol extension to avoid class hierarchy.

Reply:
That's a great idea. It'll make it easier for clients to integrate without duplicating as much code.